include/dali/core/version_util.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_VERSION_UTIL_H_
+#define DALI_CORE_VERSION_UTIL_H_
+
+namespace dali {
```

Comment:
The contents of this file are (mostly) copied from `core/util.h` - these functions are not generic enough to warrant a place in `util.h`. The differences:
* `GetVersionNumber` renamed to `MakeVersionNumber`
* Added a default value for `patch`
* Made `value` a template `V *` instead of fixed `int *`.
* Added a customizable `invalid_value`